okhttp/src/jvmTest/java/okhttp3/WebPlatformToAsciiTest.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.TestFactory
+
+/** Runs the web platform ToAscii tests. */
+class WebPlatformToAsciiTest {
+  val knownFailures = setOf(
+    // OkHttp rejects empty labels.
```

Comment:
It's hard with the failures separated from the test file? Is there a way to put the failures into the same file? Some tags, or similar?

Reply:
As-is I’m reluctant to change the content of the JSON file, since it’s from the web platform project. Hopefully not relevant for long if we can get these all passing.